sklearn_pandas/pipeline.py

```diff
@@ -3,12 +3,37 @@
 from sklearn.utils import tosequence
 
 
+def _call_fit(fit_method, X, y=None, **kwargs):
+    """
+    helper function, calls the fit or fit_transform method with the correct
+    number of parameters
+
+    fit_method: fit or fit_transform method of the transformer
+    X: the data to fit
+    y: the target vector relative to X, optional
+    kwargs: any keyword arguments to the fit method
+
+    return: the result of the fit or fit_transform method
+
+    WARNING: if this function raises a TypeError exception, test the fit
+    or fit_transform method passed to it in isolation as _call_fit will not
+    distinguish TypeError due to incorrect number of arguments from
+    other TypeError
+    """
+    try:
+        return fit_method(X, y, **kwargs)
+    except TypeError:
```

Comment:
This is pretty clever, I hadn't thought of doing something like this before. Unfortunately when I tested it, the error message accompanying a TypeError varies between Python 2 and 3. On 2, it's "test_func() takes exactly 2 arguments (1 given)" but on 3 it's "test_func() missing 1 required positional argument:". It's probably safer to leave as is.